specs/AnimatedVisualPlayer Spec.md

```diff
@@ -0,0 +1,145 @@
+
+There are some performance issues with the current implementation of LottieGen and AnimatedVisualPlayer: 
+1)	Complex Lottie animations can instantiate a lot of composition animations 
+which may impact performance and overall battery life. 
+2)	Lottie files that have a lot of complex animations even when they are paused
+have some cost to performance. 
+The Idle (paused) Lottie animations use the same amount of CPU resources as
+when they are playing the actual animation.
+
+The new APIs here allow `AnimatedVisualPlayer`,
+[AnimatedIcon](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.AnimatedIcon),
+and the interface generated by LottieGen to be updated to help reduce the number of composition animations created
+when the animation is idle or paused to help improve performance when a developer uses
+Lottie Animations with `AnimatedVisualPlayer`.
+
+This spec will focus on changes for LottieGen and `AnimatedVisualPlayer`. 
+
+This will require 3 changes: 
+1. Create a new `IAnimatedVisual2` interface which adds to
+[IAnimatedVisual](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.IAnimatedVisual)
+the ability to destroy and instantiate animations.
+2. Create a new `IAnimatedVisualSource3` interface that extends
+[IAnimatedVisualSource](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.IAnimatedVisualSource)
+with the ability to return the new `IAnimatedVisual2`
+3.	Add a `CacheMode` property to `AnimatedVisualPlayer` that will control
+when animations should be destroyed and when they should be instantiated.
+This will be implemented using the previous new interfaces.
+
+# API Pages
+
+## AnimatedVisualPlayer.CacheMode property
+
+Gets or sets a value that determines how animations are cached. Defaults to 'Always'.
+
+The two modes for the property will specify the behavior the AnimatedVisualPlayer will have
+when the player is idle (when `PlayAsync` is not active).
+
+|Mode| Behavior|
+|---- | -------|
+|Always| `AnimatedVisualPlayer` will pre-create animations even before `PlayAsync()` is called,
+and not destroy any when player is idle. |
+|None | `AnimatedVisualPlayer` will not create animations until `PlayAsync()` is called,
+and will destroy them when it completes. |
+
+If you have an animation that will start on click, you may want to set `CacheMode` to `Always`
+when the mouse enters this control, 
+and back to `None` when mouse leaves to ensure that the animation starts immediately on click,
+but don't use up resources otherwise.
+
+Setting the `CacheMode` to `Always` when the mouse enters the control will ensure that
+the animation objects are preloaded so that the animation may start immediately when it needs to play. 
+
+If your player is does not have any animations to play and `CacheMode` is set to `Always`, the player will instantiate all animations immediately and play what was instantiated. 
```

Comment:
I don't understand this sentence. If there are no animations, then it instantiates the animations and then plays what it could instantiate?

1. If there are no animations, then what is there to instantiate?
2. The text suggests that the player may fail to instantiate all animations, in which case it just plays what it can. Is that true?

Maybe this is saying, "If you call PlayAsync before animations are loaded, then the player won't start until all animations are loaded completely."

Reply:
Yes what you suggested is what I was trying to communicate. I will update the spec to match what you are saying.